src/framework/components/anim/component.js

```diff
@@ -72,6 +72,50 @@ Object.assign(AnimComponent.prototype, {
             var layer = stateGraph.layers[i];
             addLayer.bind(this)(layer.name, layer.states, layer.transitions, i);
         }
+        this.setAnimationAssets();
+    },
+
+    setAnimationAssets: function () {
+        var animationAssets = {};
+        for (var i = 0; i < this.data.layers.length; i++) {
+            var layer = this.data.layers[i];
+            var layerName = layer.name;
+            for (var j = 0; j < layer.states.length; j++) {
+                var stateName = layer.states[j];
+                if (stateName !== 'START' && stateName !== 'END') {
+                    animationAssets[layerName + ':' + stateName] = {

```

Comment:
it would be more efficient to assign to `this.data.animationAssets` directly and then call `this.loadAnimationAssets();` after the loop instead. what you've done here is clean though, so if there won't be many layers and states and it's very fast anyway, then fine to leave as-is.

Reply:
👍 I've went ahead with this approach for efficiency.